client/src/main/java/io/confluent/kafka/schemaregistry/client/MockSchemaRegistryClient.java

```diff
@@ -71,7 +72,13 @@ private int getIdFromRegistry(String subject, Schema schema, boolean registerReq
       idSchemaMap = new HashMap<Integer, Schema>();
     }
     if (registerRequest) {
-      int id = ids.incrementAndGet();
+      int id;
+      if (schemaIdCache.containsKey(schema)) {
```

Comment:
if you use `Integer` instead of `int` you can get it and put it back if it's `null`